labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -47,7 +53,19 @@ object PatternMatchingExercise {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case s:String => s"A string with length ${s.length}"
+      case i:Int if i > 0 => "A positive integer"
+      case p:Person => s"A person with name: ${p.name}"
+      case Seq(_1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _*) =>
```

Comment:
You could also use an if to check the sequence's length. My corresponding solution was:
` case seq:Seq[_] if seq.length > 10 => "Seq with more than 10 elements"`